test/extensions/filters/network/local_ratelimit/local_ratelimit_fuzz_test.cc

```diff
@@ -0,0 +1,96 @@
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::NiceMock;
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace LocalRateLimitFilter {
+struct ActiveFilter {
+  ActiveFilter(const ConfigSharedPtr& config) : filter_(config) {
+    filter_.initializeReadFilterCallbacks(read_filter_callbacks_);
+  }
+
+  NiceMock<Network::MockReadFilterCallbacks> read_filter_callbacks_;
+  Filter filter_;
+};
+
+DEFINE_PROTO_FUZZER(
+    const envoy::extensions::filters::network::local_ratelimit::LocalRateLimitTestCase& input) {
+
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  } catch (const ProtobufMessage::DeprecatedProtoFieldException& e) {
+    ENVOY_LOG_MISC(debug, "DeprecatedProtoFieldException: {}", e.what());
+    return;
+  }
+  if (input.config().token_bucket().fill_interval().nanos() < 0) {
+    ENVOY_LOG_MISC(debug, "In fill_interval, nanos should not be negative!");
+    return;
+  }
+
+  try {
+    NiceMock<Event::MockDispatcher> dispatcher_;
+    Stats::IsolatedStoreImpl stats_store_;
+    NiceMock<Runtime::MockLoader> runtime_;
+    Event::MockTimer* fill_timer_ = new Event::MockTimer(&dispatcher_);
+    envoy::extensions::filters::network::local_ratelimit::v3::LocalRateLimit proto_config =
+        input.config();
+    ConfigSharedPtr config_ =
+        std::make_shared<Config>(proto_config, dispatcher_, stats_store_, runtime_);
+    ActiveFilter active_filter(config_);
+    std::chrono::milliseconds fill_interval_(
+        PROTOBUF_GET_MS_REQUIRED(proto_config.token_bucket(), fill_interval));
+
+    for (const auto& action : input.actions()) {
+      ENVOY_LOG_MISC(trace, "action {}", action.DebugString());
+
+      switch (action.action_selector_case()) {
+      case envoy::extensions::filters::network::local_ratelimit::Action::kOnData: {
+        Buffer::OwnedImpl buffer(action.on_data().data());
+        active_filter.filter_.onData(buffer, action.on_data().end_stream());
+        break;
+      }
+      case envoy::extensions::filters::network::local_ratelimit::Action::kOnNewConnection: {
+        active_filter.filter_.onNewConnection();
+        break;
+      }
+      case envoy::extensions::filters::network::local_ratelimit::Action::kRefill: {
+        EXPECT_CALL(*fill_timer_, enableTimer(fill_interval_, nullptr));
+        fill_timer_->invokeCallback();
+        break;
+      }
+      default:
+        // Unhandled actions
+        PANIC("A case is missing for an action");
+      }
+    }
+  } catch (const EnvoyException e) {
+    ENVOY_LOG_MISC(debug, "EnvoyException in fuzz test: {}", e.what());
```

Comment:
Why is an arbitrary EnvoyException OK behavior here?

Reply:
> Why is an arbitrary EnvoyException OK behavior here?

I catch EnvoyException because in
`source/extensions/filters/network/local_ratelimit/local_ratelimit.cc `
There is an EnvoyException thrown:
`  if (fill_interval_ < std::chrono::milliseconds(50)) {
    throw EnvoyException("local rate limit token bucket fill timer must be >= 50ms");
  }`

Do you have any suggestion on how to deal with this case?

Reply:
Jianwen has some history about this -- 
I think it's OK to add this constraint in the token bucket proto (which is only used in this filter), but the question came up of having 100% line coverage in the filter.

But the middleground might be just to catch the exception when creating the config

Reply:
> Jianwen has some history about this --
> I think it's OK to add this constraint in the token bucket proto (which is only used in this filter), but the question came up of having 100% line coverage in the filter.
> 
> But the middleground might be just to catch the exception when creating the config

That makes sense. Modified.